### PR TITLE
Implement the btcUsdPriceListForGraph qeury

### PR DESCRIPTION
**Pull request by twshelton**
Implement types and mutation for Price List for Graph

### CODE REVIEW

File: src/graphql/root/query/btc-usd-price-list-for-graph.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import BtcUsdPrice from "@graphql/types/object/btc-usd-price"
+import PriceGraphRange from "@graphql/types/scalar/price-graph-range"
+
+import { getHourlyPrice } from "@services/local-cache"
+import { getCurrentPrice } from "@services/realtime-price"
+
+const rangeStartDateMS: (string) => number | Error = (range) => {
+  const currentTimestamp = Date.now()
+
+  switch (range) {
```
Comment (bodymindarts) [inline src/graphql/root/query/btc-usd-price-list-for-graph.ts:14]: This switch is a good place to use static exhaustiveness guarantee

File: src/graphql/main/schema.graphql [hunk 2]
```diff
@@ -309,6 +310,22 @@ type Coordinates {
   latitude: Float!
 }
 
+type BtcUsdPrice {
+  timestamp: Timestamp!
+  price: Float!
```
Comment (bodymindarts) [inline src/graphql/main/schema.graphql:315]: Don't we have a more specific type? Like USDPerSat or something like that? `Float` is not a good type to use.

**Review by bodymindarts — CHANGES_REQUESTED** [review]
Integration tests aren't passing

File: src/graphql/main/schema.graphql (hunk 2 above)
Comment (samerbuna) [reply src/graphql/main/schema.graphql]: Changed to use a price object with base/offset

File: src/graphql/main/schema.graphql [hunk 3]
```diff
@@ -317,6 +318,39 @@ type MobileVersions {
...
+}
+
+"""
+The `SafeInt` scalar type represents non-fractional signed whole numeric values. SafeInt can represent values between -(2^53) + 1 and 2^53 - 1
+"""
+scalar SafeInt
+
+enum ExchangeCurrencyUnit {
+  BTCSAT
+  USDCENT
```
Comment (jrbuhl93) [inline src/graphql/main/schema.graphql:340]: This pr looks good to me. Wanted to highlight this for @bodymindarts, and see what he thinks about putting both the currency and unit into one string type like this.

File: src/graphql/root/query/btc-price-list.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+} from "@graphql/types/scalar/price-graph-range"
+
+import { getHourlyPrice } from "@services/local-cache"
+import { getCurrentPrice } from "@services/realtime-price"
+
+const rangeStartDateMS: (string: typeof priceRangeValues[number]) => number = (range) => {
+  const currentTimestamp = Date.now()
+
+  switch (range) {
+    case "ONE_DAY":
```
Comment (dolcalmi) [inline src/graphql/root/query/btc-price-list.ts:18]: this should have a default value

File: src/graphql/root/query/btc-price-list.ts [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+      prices.push({
+        timestamp: price.id,
+        price: {
+          formattedAmount: btcPriceInCents.toString(),
+          base: Math.round(btcPriceInCents * 10 ** 4),
+          offset: 4,
+          currencyUnit: "USDCENT",
+        },
+      })
+    }
```
Comment (dolcalmi) [inline src/graphql/root/query/btc-price-list.ts:65]: this should be a reduce (or just an splice like in old server) also.. do we have 5 years of hourly prices in cache? (I dont see the filter in getHourlyPrice)

**Review by dolcalmi — CHANGES_REQUESTED** [review]

File: src/graphql/root/query/btc-price-list.ts (hunk 4 above)
Comment (samerbuna) [reply src/graphql/root/query/btc-price-list.ts:18]: The default value is intentionally left out here for static exhaustiveness

File: src/graphql/root/query/btc-price-list.ts (hunk 5 above)
Comment (samerbuna) [reply src/graphql/root/query/btc-price-list.ts:65]: I think `for...of` with early `break` here is better until we have an option to query the price data by date.

**Review by dolcalmi — APPROVED** [review]

**Review by bodymindarts — APPROVED** [review]

**Pull request merged**